src/application/mod.rs

```diff
@@ -105,17 +107,27 @@ impl Default for Application {
 
 impl Application {
     /// Creates a new Application using the given ApplicationConf
+
     fn new(conf: ApplicationConf) -> Self {
-        #[cfg(feature = "metrics")]
+        #[cfg(all(feature = "prometheus_exporter", feature = "metrics", not(test)))]
         {
             let _builder = PrometheusBuilder::new();
```

Comment:
if we are not storing the metrics recorder/builder ``PrometheusBuilder`` then just install directly without variables.

```rust
PrometheusBuilder::new()
  .install()
  .expect("failed to install Prometheus recorder");
```